ofrak_core/ofrak/ofrak_context.py

```diff
@@ -158,3 +168,22 @@ def _setup(self):
             self.discover(ofrak_components)
         except ModuleNotFoundError:
             pass
+
+
+class _ComponentDepAuditor:
```

Comment:
Adding a class for this seems unnecessary, I found it slower to see what was going on than a simple `if not all(dependency.is_tool_installed() for dependency in component.external_dependencies)` check in the loop over all components would be. If we really want to keep such mechanics out of the `create_ofrak_context` method, then I think adding an `dependencies_are_installed` method to `ComponentInterface`/`AbstractComponent` makes more sense.

Reply:
I'm not sure about this, new class does seem unnecessary, but from a user perspective, I could see something similar being used to clean up some DI stuff that is confusing to users, I.E conflicting/duplicated/stale components.